### ISSUE TITLE
Failure when checking org.freedesktop.fwupd.json

### ISSUE THREAD

**Issue by hughsie** [flathub/flatpak-builder-lint#23]
I just get a failed build and:

	{
	    "errors": [
		"toplevel-cleanup-debug",
		"flathub-json-skip-appstream-check"
	    ],
	    "warnings": [
		"toplevel-unecessary-branch",
		"toplevel-command-is-path"
	    ]
	}

This isn't super helpful; I literally don't know what to do.

**Comment by hfiguiere** [comment]
> I just get a failed build and:
> 
> ```
> {
>     "errors": [
> 	"toplevel-cleanup-debug",

there is `/lib/debug` in cleanup. it shouldn't be this prevent to have proper debug info package generate.

> 	"flathub-json-skip-appstream-check"

appstream check in flathub.json. If justified, an exception needs to be added.

>     ],
>     "warnings": [
> 	"toplevel-unecessary-branch",

Don't set `branch` in the manifest for `stable`.

> 	"toplevel-command-is-path"

`command` shouldn't be a path., but a command that will be found in the path.

**Comment by hfiguiere** [comment]
the last two are warnings.

**Comment by hughsie** [comment]
Right, but it says none of that in the failure in flathub....

**Comment by hughsie** [comment]
> appstream check in flathub.json. If justified, an exception needs to be added.

fwupd is a command line app, and I don't think Flathub wants to show CLI apps. Is this not true anymore?

**Comment by hfiguiere** [comment]
> > appstream check in flathub.json. If justified, an exception needs to be added.
> 
> fwupd is a command line app, and I don't think Flathub wants to show CLI apps. Is this not true anymore?

This is a valid exception, definitely.

**Comment by hughsie** [comment]
Are not all console apps on flathub valid exceptions?

**Comment by hfiguiere** [comment]
they are definitely hidden, but they also can have a valid appstream file.

**Comment by hfiguiere** [comment]
See #24

**Comment by hughsie** [comment]
ohh, fwupd has a valid metainfo file

**Comment by hughsie** [comment]
If it has a metainfo.xml, do i still need the flathub.json?

**Comment by hfiguiere** [comment]
If it validates, no you don't need the exception. That's the best option

**Comment by hughsie** [comment]
> If it validates...

https://github.com/flathub/org.flatpak.Builder/pull/111 would fix the validation.

**Comment by hfiguiere** [comment]
is this issue still a problem? Or can we just close it now.
